### PR TITLE
adding Circle

### PR DESCRIPTION
**Pull request by Jaidjen**

### CODE REVIEW

**Comment by natasha-aleksandrova** [comment]
@charlesrobison there is now trigrams assignment part of this PR - can you please review it as well.

**Comment by charlesrobison** [comment]
Hi @Jaidjen - just took a look at trigrams - doesn't look like it's ready for review.  Let us know if you have questions or need help.  When complete, please submit for review in a new PR

Ready to merge

**Comment by Jaidjen** [comment]
Hello,

I worked with Natasha on this yesterday and I thought it was ready? It’s not showing again? 
Do I need to resend?

Thank you,
Diana

**Comment by charlesrobison** [comment]
@Jaidjen - unfortunately this doesn't meet the assignment requirements.  From the assignment text, "For this exercise, try implementing a trigram algorithm that generates a couple of hundred words of text using a book-sized file as input."  You'll want to be able to feed in text, like a book from Project Gutenberg, and use it to create new text, using those words from the book.

File: students/Jaidjen/lesson04/Trigrams.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    trigrams = dict()
+    trigrams['I wish'] = ['I']
+    trigrams['wish I'] = ['may']
+    trigrams['I may'] = ['I']
+    trigrams['may I'] = ['wish']
+    trigrams['I wish'].append('I')
+    trigrams['wish I'].append('might')
+
+    print("This is a trigrams")
+    return trigrams
```
Comment (natasha-aleksandrova) [inline students/Jaidjen/lesson04/Trigrams.py]: you're not using `words` in here - if this function was implemented to create trigrams from words list then you could have easily updated to feed in data from a a file.

File: students/Jaidjen/lesson04/Trigrams.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    trigrams = build_trigrams(words)
+    new_text=[]
+    start_key=random.choice(list(trigrams))
+    new_text.append(start_key)
+    print("This is the new text")
+    print(new_text)
+    print("This is the start key")
+    print(start_key)
+    for i in range(len(trigrams)):
+        random.choice(list(trigrams))
```
Comment (natasha-aleksandrova) [inline students/Jaidjen/lesson04/Trigrams.py]: this function doesn't appear to create new words

File: students/Jaidjen/lesson04/Trigrams.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+
+
+if __name__== "__main__":
+      trigrams = build_trigrams(words)
+      print(trigrams)
+      make_words = word_pair()
+      print(words)
+      new_words()
```
Comment (natasha-aleksandrova) [inline students/Jaidjen/lesson04/Trigrams.py]: from the tutorial:
```
in_data = read_in_data(filename)  
words = make_words(in_data)
word_pairs = build_trigram(words)
new_text = build_text(word_pairs)

```
the last 2 functions from the above is where your main logic would go.

currently your build_trigrams just has hardcoded dictionary and I don't see anything equivalent to build_text function.

**Comment by Jaidjen** [comment]
Hello,

Ok I know it’s probably too late to ask this question but I just saw this email. I read and read the instructions and I can’t figure out how to do that. Is there another resource I can look at to see how this works? I’m trying to envision it and I can’t and I think that’s part of the problem. 

Thank you!

**Comment by natasha-aleksandrova** [comment]
@Jaidjen 

I would tackle the problem one small section at a time.

work on only implementing:
`build_trigrams` function.

https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/kata_fourteen.html#building-the-trigrams-dict

this tutorial walks you through step by step to write the code, including providing the code logic to create words and followers.

are there specific questions on this function that you have?

**Comment by Jaidjen** [comment]
I am trying to figure out how to feed text from a book to build the trigrams. I will go over the tutorial again. 

Thank you!

**Comment by natasha-aleksandrova** [comment]
before you even try to feed in the text from the book (which is the easier part), your core function build_trigrams needs to be implemented first.

**Pull request merged**